src/components/main/Main.js

```diff
@@ -31,25 +31,41 @@ function Main() {
     setCards([...cards, newCard]);
   };
 
+  const updateCard = (updatedCard) => {
+    const newCards = cards.map((card) => {
+      if (card.id === updatedCard.id) {
+        return updatedCard;
+      }
+      return card;
+    });
+    setCards([...newCards]);
+  };
+
+  const removeCard = (cardId) => {
+    const filtered = cards.filter((card) => card.id !== cardId);
+    setCards([...filtered]);
```

Comment:
The filter() method creates a new array, so why are you recreating it here?

Reply:
fixed